src/solvers/dg_multi/fluxdiff.jl

```diff
@@ -0,0 +1,182 @@
+
+  # temp storage for entropy variables at volume quad points
+  entropy_var_values = StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(rd.Nq, md.num_elements), nvars))
+
+  # local storage for interface fluxes, rhs, and source
+  flux_face_values = similar(u_face_values)
+  rhs_local_threaded = [StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(num_quad_points_total), nvars)) for _ in 1:Threads.nthreads()]
+  local_values_threaded = [StructArray{SVector{nvars, uEltype}}(ntuple(_->zeros(rd.Nq), nvars)) for _ in 1:Threads.nthreads()]
+
+  return (; md, Qrst_skew_Tr, VhP, Ph, invJ = inv.(md.J),
+      entropy_var_values, projected_entropy_var_values, entropy_projected_u_values,
+      u_values, u_face_values, rhs_local_threaded, flux_face_values, local_values_threaded)
+end
+
+function entropy_projection!(cache, u::StructArray, mesh::VertexMappedMesh,
+                             equations, dg::DG) where {DG <: PolyDGFluxDiff}
+
+  rd = dg.basis
+  @unpack Vq = rd
+  @unpack VhP, entropy_var_values, u_values, entropy_var_values = cache
+  @unpack projected_entropy_var_values, entropy_projected_u_values = cache
+
+  # TODO: simplices. Address hard-coding of `entropy2cons!`
+  StructArrays.foreachfield(mul_by!(Vq), u_values, u)
+  entropy_var_values .= cons2entropy.(u_values, equations)
+
+  # "VhP" fuses the projection "P" with interpolation to volume and face quadrature "Vh"
+  StructArrays.foreachfield(mul_by!(VhP), projected_entropy_var_values, entropy_var_values)
+  entropy_projected_u_values .= entropy2cons.(projected_entropy_var_values, equations)
+end
+
+function calc_volume_integral!(du, u::StructArray, volume_integral::VolumeIntegralFluxDifferencing,
+                               mesh::VertexMappedMesh, equations, dg::PolyDGFluxDiff, cache)
+
+  rd = dg.basis
+  md = mesh.md
+  @unpack entropy_projected_u_values, rhs_local_threaded, Ph = cache
+  volume_flux_oriented(i) = @inline (u_ll, u_rr)->volume_integral.volume_flux(u_ll, u_rr, i, equations)
+
+  # skips subblock of Qi_skew_Tr which we know is zero by construction
+  skip_index(i,j) = i > rd.Nq && j > rd.Nq
+
+  # Todo: simplices. Dispatch on curved/non-curved mesh types, this code only works for affine meshes (accessing rxJ[1,e],...)
+  @threaded for e in eachelement(mesh, dg, cache)
+    rhs_local = rhs_local_threaded[Threads.threadid()]
+    fill!(rhs_local, zero(eltype(rhs_local)))
+    u_local = view(entropy_projected_u_values, :, e)
+    for i in eachdim(mesh)
+      Qi_skew_Tr = build_lazy_physical_derivative(e, i, mesh, dg, cache)
+      hadamard_sum_ATr!(rhs_local, Qi_skew_Tr, volume_flux_oriented(i), u_local, skip_index)
+    end
+    StructArrays.foreachfield(mul_by_accum!(Ph), view(du, :, e), rhs_local)
+  end
+end
+
+function rhs!(du, u::StructArray, t, mesh, equations,
+              initial_condition, boundary_conditions::BC, source_terms::Source,
+              dg::PolyDGFluxDiff, cache) where {BC, Source}
```

Comment:
```suggestion
function rhs!(du, u, t, mesh, equations, initial_condition, boundary_conditions,
              source_terms, dg::PolyDGFluxDiff, cache)
```
Same as above - are the extra annotations/parameters really necessary?